src/puppetlabs/puppetdb/command.clj

```diff
@@ -235,8 +235,9 @@
 (defn store-report*
   [version db {:keys [payload annotations]}]
   (let [id          (:id annotations)
-        report      (upon-error-throw-fatality
-                     (report/validate! version payload))
+        report      (->> payload
+                         (s/validate report/full-report)
```

Comment:
this is getting validated twice on the way in, once in command and immediately afterward in storage. I can see that we're also doing that with some of the other commands, but it seems unnecessary. I'm not sure that it matters where the validation goes; all things equal I think storage is probably better because then it gets hit by a lot more test code.